packages/compiler-core/__tests__/transforms/transformElement.spec.ts

```diff
@@ -1224,4 +1224,55 @@ describe('compiler: element transform', () => {
       isBlock: false
     })
   })
+  
+  describe('<KeepAlive> multiple children', () => {
+
+    function checkWarning(
+      template: string,
+      shouldWarn: boolean,
+      message = '<KeepAlive> expects exactly one child component.'
+    ) {
+      const spy = jest.fn()
+
+      parseWithBind(template.trim(), {
+        onError: err => {
+          spy(err.message)
+        }
+      })
+
+      if (shouldWarn) expect(spy).toHaveBeenCalledWith(message)
+      else expect(spy).not.toHaveBeenCalled()
+    }
+
+    test('does not warn if has one children', () => {
+      checkWarning(
+        `<KeepAlive>
+           <component :is="activeComponent"/>
+         </KeepAlive>`,
+        false
+      )
+    })
+
+    test('warn if has multiple children', () => {
+      checkWarning(
+        `<KeepAlive>
+          <component :is="activeComponent"/>
+          <component :is="activeComponent2"/>
+         </KeepAlive>`,
+        true
+      )
+    })
+
+    test('should ignore comments', () => {
```

Comment:
I think there should be `warn if` tests like above but with comments to verify that comments don't change the result. They would be similar to this test (`should ignore comments`) except they're verifications for the warning.